### PR TITLE
Copy csproj reference assemblies to build output

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/dotnet/cli/issues/2961

### CODE REVIEW

File: src/Microsoft.DotNet.Compiler.Common/LibraryExporterExtensions.cs [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
-using Microsoft.DotNet.ProjectModel;
 using Microsoft.DotNet.ProjectModel.Compilation;
 using Microsoft.DotNet.ProjectModel.Graph;
-using Microsoft.Extensions.DependencyModel;
 
 namespace Microsoft.DotNet.Cli.Compiler.Common
 {
     public static class LibraryExporterExtensions
     {
+        public static IEnumerable<LibraryExport> GetProjectDependencies(this LibraryExporter exporter)
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Compiler.Common/LibraryExporterExtensions.cs]: I am not a big fan of this name. It is confusing to me if we are getting all the dependencies that the project depends on or all the dependencies of type project.

Maybe GetAllProjectTypeDependencies?

**Comment by livarcocc** [comment]
Add some tests.

**Comment by pakrym** [comment]
Yeah, trying to make tests work without VS

File: src/Microsoft.DotNet.Compiler.Common/LibraryExporterExtensions.cs (hunk 1 above)
Comment (pakrym) [inline src/Microsoft.DotNet.Compiler.Common/LibraryExporterExtensions.cs]: OK

**Comment by pakrym** [comment]
@livarcocc any ideas on how repro this bug in test without VS and committing lock file?

**Comment by livarcocc** [comment]
@pakrym I see the problem. I don't think we have an easy way to do that today, unfortunately. Ideally, we should be able to instantiate LockFiles with the configuration we wanted without depending on actually reading a lock file. I am fine to check-in like this if you managed to validate it against VS.

:shipit:

**Pull request merged**